lib/remote-objects.js

```diff
@@ -160,12 +160,46 @@ RemoteObjects.prototype.invoke = function(method, ctorArgs, args, callback) {
 
 /**
  * Get an adapter by name.
- * @param {String} name The adapter name
+ * @param {String} nameOrClass The adapter name or provided class
  * @return {Adapter}
  */
 
-RemoteObjects.prototype.adapter = function(name) {
-  return require('./' + name + '-adapter');
+RemoteObjects.prototype.adapter = function(nameOrClass) {
+  if (typeof nameOrClass === 'function') {
+    // Throws an error if the provided adapter doesn't meets the requirements
+    this.validateAdapter(nameOrClass);
+
+    return nameOrClass;
+  } else {
+    return require('./' + nameOrClass + '-adapter');
+  }
+};
+
+RemoteObjects.prototype.getRequiredAdapterMethods = function() {
+  return ['createHandler'];
+};
+
+RemoteObjects.prototype.validateAdapter = function(Adapter) {
+  var requiredMethods = this.getRequiredAdapterMethods();
+
+  var notImplementedMethods = requiredMethods.reduce(
+      function(notImplementedMethods, methodString) {
+        if (typeof Adapter.prototype[methodString] !== 'function') {
+          return notImplementedMethods.concat(methodString);
+        }
+
+        return notImplementedMethods;
+      }, []);
```

Comment:
I think this can be simplified by using `filter`:

```js
const notImplementedMethods = requireMethods.filter(
  methodString => typeof Adapter.prototype[methodString] !== 'function')
```

Note that since 3.0 release, we target Node 4+, thus we can (and should be) using ES6 constructs like `const` and arrow functions.

See also:
 - http://loopback.io/doc/en/contrib/style-guide.html#variable-declarations
 - http://loopback.io/doc/en/contrib/style-guide.html#arrow-functions